doc/wg/core/notes/core-notes-2021-07-02.md

```diff
@@ -0,0 +1,333 @@
+  answer is "components make it easier, hopefully".
+* Amit: Right, ideally this is something that is resolved in the board
+  configuration. We could make the board configuration do this automatically.
+* Alexandru: I don't see how. You have the `with_driver` function that has
+  nothing to do with the components. Then you supply the driver number to the
+  component, so I don't think components solve the problem.
+* Amit: I agree components on their own don't. My dream for several years is a
+  macro for specifying a driver that compiles into the spaghetti code we have
+  now.
+* Brad: This is a fundamental issue to both designs, choosing v3 or v4 does not
+  address it.
+* Hudson: Sorry, I joined a little late. The problem you are discussing is if a
+  different driver number is passed to `with_driver` and to `create_grant` that
+  you will get potentially cryptic and weird bugs? For v4 the check is currently
+  at runtime and it just returns an error.
+* Brad: Correct
+* Hudson: Could we panic when that happens? Then it would be easy to debug.
+* Brad: Then you would be panicing the kernel.
+* Hudson: I think it is okay, because it is trusted code in `main.rs`. A capsule
+  can't do it.
+* Phil: How would this work? Where would this panic check go?
+* Hudson: When Subscribe into a grant it passes the driver num it is expecting
+  the grant to have and if they don't match, there's a panic.
+* Brad: Wait, when?
+* Hudson: Where we currently return an error in the code.
+* Brad: Sorry, I'm not quite sure. Is a capsule triggering this, is the kernel
+  triggering this?
+* Hudson: No, it would be in the kernel code.
+* Brad: Right, but I'm still not following when. What series of events have
+  happened?
+* Hudson: The series of events is a capsule has called subscribe.
+* Brad: Capsules don't call subscribe, processes do.
+* Hudson: Yeah, so a process has called subscribe.
+* Brad: But then how does the kernel check?
+* Hudson: I added this check yesterday...
+* Brad: Let's talk about this offline.
+
+## Code size reduction effort update
+* Hudson: I sent a PR which I don't expect to be merged until after the 2.0
+  stuff settles. The high-level takeaway for the PR is we have a lot of
+  high-level functionality in methods on `Grant` that are generic over the type
+  in the Grant and/or closure. E.g. every time you call `Grant::enter` you
+  duplicate the logic in the binary. It's generally inlined into the use site,
+  and for boards like Imix that means 50 or more copies of the identical code.
+  This PR moves the portions of the functions that do not rely on the generic
+  parameters into non-generic functions that are marked `#[inline(never)]`. I
+  then call those from the generic functions. For code that doesn't need to be
+  duplicated for the generic types, you only have a single instance of those
+  function. As written, this saved about 7300 bytes on Imix. Functionallity
```

Comment:
```suggestion
  function. As written, this saved about 7300 bytes on Imix. Functionally it is
```

Reply:
Fixed.

(Sorry everyone for the amend + force-push, that was accidental).